### PR TITLE
Classifier equipped with adversarial network

### PR DESCRIPTION
**Pull request by suraz09**
Added all the functionality with the adversarial network.
Mitigates the bias of the unfair classifier

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 **Machine Learning** models are extensively used for the purpose of determining access to services such as eligibility for loan, insurance etc. Despite gains in efficiency of these models, unintentional discrimination flaw of the model has not been fully addressed. 
 
-The Fair Classifier is an open source package that evaluates the fairness of a Neural Network based on the data it has been trained on.
+The Fair Classifier is an open source package that evaluate the fairness of a Neural Network. The goal is to build a command-line playground where user can check the fairness of the model by removing different sensitive attribute from the training data.
```
Comment (Mitchellpkt) [inline README.md]: Clearer 👍 Tiny typo, 'attribute' should be plural

File: README.md [hunk 2]
```diff
@@ -1 +1,48 @@
-# FairClassifier
\ No newline at end of file
+# FairClassifier
+
+**Machine Learning** models are extensively used for the purpose of determining access to services such as eligibility for loan, insurance etc. Despite gains in efficiency of these models, unintentional discrimination flaw of the model has not been fully addressed.
```
Comment (Mitchellpkt) [inline README.md]: Insignificant typo (loan --> plural)

File: README.md [hunk 3]
```diff
@@ -1 +1,48 @@
-# FairClassifier
\ No newline at end of file
+# FairClassifier
+
+**Machine Learning** models are extensively used for the purpose of determining access to services such as eligibility for loan, insurance etc. Despite gains in efficiency of these models, unintentional discrimination flaw of the model has not been fully addressed. 
+
+The Fair Classifier is an open source package that evaluate the fairness of a Neural Network. The goal is to build a command-line playground where user can check the fairness of the model by removing different sensitive attribute from the training data.
```
Comment (Mitchellpkt) [inline README.md]: Insignificant typo (attribute --> plural)

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+ 
+This python package includes a set of metrics to test models for biasness and use adversarial network to help mitigate that biasness of the training data set.
```
Comment (Mitchellpkt) [inline README.md]: Does this handle both **model bias** and **training data set bias** or are these considered equivalent?

File: README.md [hunk 5]
```diff
@@ -1 +1,48 @@
...
+
+
+### Fairness metrics implemented
+> The fairness contraint P% rule is based on this paper [Link.](https://arxiv.org/pdf/1507.05259.pdf)
```
Comment (Mitchellpkt) [inline README.md]: Insignificant comment, I would do `P% rule is based on [this paper](https://....).` but that's just stylistic

File: src/model/classifier.py [hunk 6]
```diff
@@ -1,5 +1,9 @@
 from keras.layers import Input, Dense, Dropout
 from keras.models import Model
+import sys, os
+
+sys.path.append(os.path.join(os.path.dirname(os.path.dirname(sys.path[0])), 'configs' ))
```
Comment (Mitchellpkt) [inline src/model/classifier.py:5]: 👍

File: src/preprocess/preprocess.py [hunk 7]
```diff
@@ -17,25 +17,40 @@ def __init__(self, file):
...
 
-        return X, y, Z
 
     """
     Split the data into train and test set.
     """
-    def split_data(self,X, y, Z):
+    def split_data(self, X, y, Z):
         X_train, X_test, y_train, y_test, Z_train, Z_test = train_test_split(X, y, Z, test_size=0.3, stratify=y,
                                                                              random_state=7)
```
Comment (Mitchellpkt) [inline src/preprocess/preprocess.py]: Is it good idea to leave `random_state` as specified in non-example code? I see that often in examples, so the reader can generate the same plots as the article, etc. Do you want to set the random state, in this context? 

I'm not 100% sure of conventions, so I'm not suggesting either way.

**Review by Mitchellpkt — APPROVED** [review]
Looks good 👍 

Made a few insignificant comments about grammar. 

Code looks good to me! Approving.

File: README.md (hunk 2 above)
Comment (suraz09) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (suraz09) [reply README.md]: I meant bias of the model, fixed it by changing it to fairness, others might also be confused by my wording here so.

File: src/model/classifier.py (hunk 6 above)
Comment (suraz09) [reply src/model/classifier.py:5]: :)

File: src/preprocess/preprocess.py (hunk 7 above)
Comment (suraz09) [reply src/preprocess/preprocess.py]: yes, thank you for noticing, removed it.

File: README.md (hunk 5 above)
Comment (suraz09) [reply README.md]: done

**Pull request merged**